bundles/org.openhab.binding.ambientweather/src/main/java/org/openhab/binding/ambientweather/internal/handler/AmbientWeatherBridgeHandler.java

```diff
@@ -0,0 +1,233 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ambientweather.internal.handler;
+
+import static org.openhab.binding.ambientweather.internal.AmbientWeatherBindingConstants.CONFIG_MAC_ADDRESS;
+
+import java.io.IOException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.ambientweather.internal.config.BridgeConfig;
+import org.openhab.binding.ambientweather.internal.model.DeviceJson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link AmbientWeatherBridgeHandler} is responsible for handling the
+ * bridge things created to use the Ambient Weather service.
+ *
+ * @author Mark Hilbush - Initial Contribution
+ */
+@NonNullByDefault
+public class AmbientWeatherBridgeHandler extends BaseBridgeHandler {
+    // URL to retrieve device list from Ambient Weather
+    private static final String DEVICES_URL = "https://api.ambientweather.net/v1/devices?applicationKey=%APPKEY%&apiKey=%APIKEY%";
+
+    // Timeout of the call to the Ambient Weather devices API
+    public static final int DEVICES_API_TIMEOUT = 20000;
+
+    // Time to wait after failed key validation
+    public static final long KEY_VALIDATION_DELAY = 60L;
+
+    private final Logger logger = LoggerFactory.getLogger(AmbientWeatherBridgeHandler.class);
+
+    // Job to validate app and api keys
+    @Nullable
+    private ScheduledFuture<?> validateKeysJob;
+
+    // Application key is granted only by request from developer
+    @Nullable
+    private String applicationKey;
+
+    // API key assigned to user in ambientweather.net dashboard
+    @Nullable
+    private String apiKey;
+
+    // Used Ambient Weather real-time API to retrieve weather data
+    // for weather stations assigned to an API key
+    private AmbientWeatherEventListener listener;
+
+    private final Gson gson = new Gson();
+
+    private Runnable validateKeysRunnable = new Runnable() {
+        @Override
+        public void run() {
+            logger.debug("Validating application and API keys");
+
+            String response = null;
+            try {
```

Comment:
You could just join the code of two try-catches so that structure is clearer, you won't need the returns you can join the response declaration with the assignment.

Reply:
Yes, much simpler indeed.